addons/point_of_sale/models/pos_session.py

```diff
@@ -408,6 +410,15 @@ def _create_account_move(self):
                         stock_expense[exp_key] = self._update_amounts(stock_expense[exp_key], {'amount': amount}, move.picking_id.date)
                         stock_output[out_key] = self._update_amounts(stock_output[out_key], {'amount': amount}, move.picking_id.date)
 
+            # Setting current partner's customer_rank to 1 if necessary
+            try:
+                if order.partner_id.id:
+                    with self.env.cr.savepoint():
+                        self.env.cr.execute("UPDATE res_partner SET customer_rank=1 WHERE ID=%s AND customer_rank=0", (order.partner_id.id,))
```

Comment:
It's not a good idea to force the customer_rank to 1. If the customer alredy exists and have a customer_rank bigger then 1 it would came back to 1.

See the original code : https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_move.py#L2126

And maybe in the code, refer to this part of the code.

Reply:
updated accordingly